File: app/src/main/java/com/bsas/androiddevs/manejoerrores/repository/db/dao/BaseDao.java
package com.bsas.androiddevs.manejoerrores.repository.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;

import com.bsas.androiddevs.manejoerrores.repository.db.entity.MovieDb;

import java.util.List;

@Dao
public interface BaseDao<T> {

    @Delete
    void delete(T entity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void upsert(T entity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void upsertAll(List<MovieDb> movieDbs);

}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/repository/db/dao/BaseDao.kt
package com.bsas.androiddevs.manejoerrores.repository.db.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy

@Dao
interface BaseDao<T> {

    @Delete
    suspend fun delete(entity: T)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsert(entity: T)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsertAll(movieDbs: List<T>)

}